realm/src/main/java/io/realm/internal/SharedGroup.java

```diff
@@ -81,34 +82,34 @@ public SharedGroup(String databaseFile, Durability durability, boolean fileMustE
         checkNativePtrNotZero();
     }
 
-    private native long createNativeWithImplicitTransactions(long nativeReplicationPtr, int durability, byte[] key);
-
-    private native long nativeCreateReplication(String databaseFile, byte[] key);
-
-    void advanceRead() {
+    /**
+     * Advances the transaction the to latest available data.
+     *
+     * @return {@code true} if version was advanced, {@code false} if already at the latest version.
+     */
+    boolean advanceRead() {
```

Comment:
if `nativeGetVersionID()` is always called after `nativeAdvanceRead()` and we currently don't expect it otherwise in the near future, I would suggest to let `nativeAdvanceRead()` return the VersionID instead to save a JNI call.